sdks/python/apache_beam/ml/inference/api.py

```diff
@@ -38,6 +38,9 @@ class PredictionResult:
 @beam.typehints.with_output_types(Union[PredictionResult, Tuple[_K, PredictionResult]])  # pylint: disable=line-too-long
 class RunInference(beam.PTransform):
   """
+  NOTE: This API and its implementation are under development and
+  do not promise backward compatibility.
```

Comment:
nit: "provide backward compatibility guarantees"

Should we add this note directly on the implementations too?

Reply:
Got sidetracked with an important project. I'd hoped to get this into the release version.